### DISCUSSION TITLE
auth flow in popup

### DISCUSSION THREAD

**Discussion by ghost** [ory/hydra#3264] [Q&A]
Hello everyone,

I'm trying to work on a MS Teams integration for our webapp where the user can login with Google. MS Teams runs our application into an iframe, which obviously doesn't allow any external login, so I tried their SDK and encountered [this](https://github.com/OfficeDev/microsoft-teams-library-js/issues/1319) kind of error. 
So, is it possible to have the auth flow opening a popup to login with third party instead of redirecting in the same window? 
I'm not an expert in this field, I've tried looking for an answer but found nothing in regard.

Many thanks

**Comment by vinckr** [comment]
Hello @dcgenius 
can you help me understand your current and desired flow please? 
a rough diagram how your auth flow works now and how it should work with MS Teams would be ideal.

**Reply by ghost** [reply]
I've got something like this, pardon me, but I just followed the redirects on my browser: ![image](https://user-images.githubusercontent.com/107853260/192619463-e4c9dda5-ed21-4a72-ac43-f7b9ee9593db.png)

What I'd like to achieve is that the redirect to the google login page should happen in a popup:
client -> login with google -> opens popup to insert credentials where it says "the application X will access Y,Z" -> popup closes on successful login -> the original window will redirect back to home page.

Now that I'm writing it, probably it's something related to client side implementation rather than backend... 
Well, this got me thinking.
